### PR TITLE
Prototypist Requisition "Reverse-Engineer" Update

### PR DESCRIPTION
**Pull request by Kubius**
[FEAT]
## About the PR 

Adds a new client to the Prototypist requisition with three new rewards, and adds a new reward to an existing client.

- The new **Artifact reverse-engineer** client has exclusively proprietary technology rewards, but the unusual condition of requiring one of a certain spread of handheld artifacts as a condition of the contract.
- The **terminus drive** is provided in a set of three, and provides a powerful new storage option; while there is a finite time each terminus can remain powered, and no items within are accessible without power, **every terminus shares the same storage**, and their shared storage space can hold up to medium-weight items. Further terminus acquisitions, while unlikely, will share the same storage.
gifber
- The **APS-4 bio-integrator** (less formally "medimulcher"), while powered, is capable of digesting almost any solid food and gradually converting it into medically-useful, topically-applicable pustules containing a small quantity of synthflesh. *Don't poke it. You are digestible.*
mulchismo2
- The **portable recharge bay** is a backpack with benefits; using an onboard large power cell (not included), it can recharge any small-cell-containing item you've loaded inside of it. As a drawback, it's much slower than standard rechargers, only has six slots for items, and won't recharge energy guns (though certain enterprising individuals may find a way around the latter) - it makes a great carry-along, though!
bpshutbig
- Additionally, a new item has been introduced for the engineering firm client. The **sonic showerhead** comes in a set (two or three), and while it eats a bit more power than your usual showerhead, it'll scrub you clean without all the de-garbing and messy water a regular shower entails. (Since this is niche, you'll also receive a little more cash as well.)

## Why's this needed? 

Additional prototypist requisition variety makes for a more appealing lineup of rewards for interacting with the requisition system.

## Changelog 

```changelog
(u)Kubius
(+)The Prototypist requisition has four new outcomes.
```

### CODE REVIEW

File: code/obj/machinery/medimulcher.dm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		src.add_fingerprint(user)
+
+		if (src.digestion_progress >= MDM_CYST_COST && !(status & (BROKEN|NOPOWER)))
+			src.digestion_progress -= MDM_CYST_COST
+			var/obj/item/I = new /obj/item/reagent_containers/patch/synthetic
+			playsound(src.loc, 'sound/impact_sounds/Glub_2.ogg', 80, 1)
+			boutput(user, SPAN_NOTICE("[src] dispenses a small glob.[prob(10) ? " Grody." : null]"))
+			user.put_in_hand_or_drop(I)
+		else
+			boutput(user, SPAN_ALERT("[src] doesn't respond to your touch."))
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:68]: a very hungry mulcher should try to eat your arm 😈

File: code/obj/machinery/shower.dm [hunk 2]
```diff
@@ -110,4 +110,99 @@
...
+
+	proc/spray()
+		var/cleaned_a_nerd = FALSE
+		src.last_spray = world.time
+
+		for (var/atom/A in range(0, get_turf(src)))
+			if (A == src) continue
+			if (ismob(A))
+				var/mob/M = A
+				if (!isdead(M))
```
Comment (Tarmunora) [inline code/obj/machinery/shower.dm:193]: should probably clean_forensic non-mobs as well?

File: code/obj/machinery/shower.dm [hunk 3]
```diff
@@ -110,4 +110,99 @@
...
+					M.clean_forensic()
+					M.delStatus("marker_painted")
```
Comment (Tarmunora) [inline code/obj/machinery/shower.dm:195]: should probably affect hygiene motive

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/machinery/shower.dm (hunk 2 above)
Comment (Kubius) [reply code/obj/machinery/shower.dm:195]: It is now thus

File: code/obj/machinery/shower.dm (hunk 3 above)
Comment (Kubius) [reply code/obj/machinery/shower.dm:197]: Did the thing (thought clean_forensic had that built in, for some reason).

File: code/obj/machinery/medimulcher.dm (hunk 1 above)
Comment (Kubius) [reply code/obj/machinery/medimulcher.dm]: It now does, if you're impolite to it.

File: code/obj/machinery/medimulcher.dm [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+		src.add_fingerprint(user)
+
+		if (src.digestion_progress >= MDM_CYST_COST && !(status & (BROKEN|NOPOWER)))
+			src.digestion_progress -= MDM_CYST_COST
+			var/obj/item/I = new /obj/item/reagent_containers/patch/synthetic
+			playsound(src.loc, 'sound/impact_sounds/Glub_2.ogg', 80, 1)
+			boutput(user, SPAN_NOTICE("[src] dispenses a small glob.[prob(10) ? " Grody." : null]"))
+			user.put_in_hand_or_drop(I)
+		else
+			if(ishuman(user) && last_poke_time && (last_poke_time + 1 SECOND > world.time))
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:71]:
```suggestion
			if(ishuman(user) && (last_poke_time + 1 SECOND > TIME))
```
also the time-based conditional seems... not useful here? esp considering you already have a cooldown below

File: code/obj/machinery/medimulcher.dm [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+				boutput(user, SPAN_ALERT("You poke [src] trying to get it to do something."))
+				if (prob(15) && !GET_COOLDOWN(src,"bit_a_nerd")) //don't keep poking at it when it's not ready to glob you.
+					var/limb_to_ouch
+					var/mob/living/carbon/human/H = user
+					limb_to_ouch = H.hand ? "l_arm" : "r_arm"
+					var/howmuchbite = rand(3,5)
+					src.stomach = min(src.stomach + howmuchbite, MDM_MAX_STOMACH)
+					H.TakeDamage(limb_to_ouch, howmuchbite*3)
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:79]: don't be a coward, take the whole arm off 😈

File: code/obj/machinery/medimulcher.dm [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+		if (src.digestion_progress >= MDM_CYST_COST && !(status & (BROKEN|NOPOWER)))
+			src.digestion_progress -= MDM_CYST_COST
+			var/obj/item/I = new /obj/item/reagent_containers/patch/synthetic
+			playsound(src.loc, 'sound/impact_sounds/Glub_2.ogg', 80, 1)
+			boutput(user, SPAN_NOTICE("[src] dispenses a small glob.[prob(10) ? " Grody." : null]"))
+			user.put_in_hand_or_drop(I)
+		else
+			if(ishuman(user) && last_poke_time && (last_poke_time + 1 SECOND > world.time))
+				boutput(user, SPAN_ALERT("You poke [src] trying to get it to do something."))
+				if (prob(15) && !GET_COOLDOWN(src,"bit_a_nerd")) //don't keep poking at it when it's not ready to glob you.
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:73]:
```suggestion
				if (prob(15) && !ON_COOLDOWN(src,"bit_a_nerd", 2 SECONDS)) //don't keep poking at it when it's not ready to glob you.
```
 no need to use GET_COOLDOWN here

File: code/obj/machinery/medimulcher.dm [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+					var/limb_to_ouch
+					var/mob/living/carbon/human/H = user
+					limb_to_ouch = H.hand ? "l_arm" : "r_arm"
+					var/howmuchbite = rand(3,5)
+					src.stomach = min(src.stomach + howmuchbite, MDM_MAX_STOMACH)
+					H.TakeDamage(limb_to_ouch, howmuchbite*3)
+					playsound(user.loc, 'sound/impact_sounds/Flesh_Crush_1.ogg', 75)
+					boutput(user, SPAN_COMBAT("[src] lashes out and gnaws on your arm! [prob(50) ? "Holy shit!" : "What the fuck?"]"))
+					ON_COOLDOWN(src,"bit_a_nerd",2 SECONDS)
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:82]:
```suggestion
```

File: code/obj/machinery/medimulcher.dm [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+			else
+				boutput(user, SPAN_ALERT("[src] doesn't respond to your touch."))
+			src.last_poke_time = world.time
```
Comment (Tarmunora) [inline code/obj/machinery/medimulcher.dm:85]:
```suggestion
			src.last_poke_time = TIME
```

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/machinery/medimulcher.dm (hunk 4 above)
Comment (Kubius) [reply code/obj/machinery/medimulcher.dm:71]: Its function is keeping you from randomly getting eaten if you just click it once without realizing it doesn't have anything to dispense, and don't repeatedly prod it trying to get it to do something.

File: code/obj/machinery/medimulcher.dm (hunk 6 above)
Comment (Kubius) [reply code/obj/machinery/medimulcher.dm:73]: Upon reexamination: yeah, for some reason I thought I had to make it separate, but I actually do not.

File: code/obj/machinery/medimulcher.dm (hunk 5 above)
Comment (Kubius) [reply code/obj/machinery/medimulcher.dm]: Added, but has probability based on how much you irritate it.

File: code/obj/machinery/medimulcher.dm (hunk 4 above)
Comment (Kubius) [reply code/obj/machinery/medimulcher.dm:74]: (Particularly important given you can lose a limb to it now.)

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**